src/service-object.ts

```diff
@@ -382,14 +402,25 @@ class ServiceObject<T = any> extends EventEmitter {
    * Set the metadata for this object.
    *
    * @param {object} metadata - The metadata to set on this object.
+   * @param {object=} options - Configuration options.
    * @param {function=} callback - The callback function.
    * @param {?error} callback.err - An error returned while making this request.
    * @param {object} callback.apiResponse - The full API response.
    */
-  setMetadata(metadata: Metadata): Promise<SetMetadataResponse>;
+  setMetadata(metadata: Metadata, options?: SetMetadataOptions):
+      Promise<SetMetadataResponse>;
+  setMetadata(
+      metadata: Metadata, options: SetMetadataOptions,
+      callback: MetadataCallback): void;
   setMetadata(metadata: Metadata, callback: MetadataCallback): void;
-  setMetadata(metadata: Metadata, callback?: MetadataCallback):
-      Promise<SetMetadataResponse>|void {
+  setMetadata(
+      metadata: Metadata,
+      optionsOrCallback?: SetMetadataOptions|MetadataCallback,
+      callback?: MetadataCallback): Promise<SetMetadataResponse>|void {
+    const options =
```

Comment:
Consider using [maybeOptionsOrCallback](https://github.com/googleapis/nodejs-common/blob/master/src/util.ts#L756) to simplify this logic!